lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex

```diff
@@ -20,21 +20,21 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ReconstructedAlert do
         %Screen{app_params: %PreFare{header: %CurrentStopId{stop_id: stop_id}}} = config,
         now \\ DateTime.utc_now(),
         fetch_routes_by_stop_fn \\ &Route.fetch_routes_by_stop/3,
-        fetch_stop_sequences_by_stop_fn \\ &RoutePattern.fetch_stop_sequences_through_stop/2,
-        fetch_alerts_fn \\ &Alert.fetch/1,
-        get_parent_station_id_fn \\ &get_parent_station/1
+        fetch_stop_sequences_by_stop_fn \\ &RoutePattern.fetch_parent_station_sequences_through_stop/2,
+        fetch_alerts_fn \\ &Alert.fetch/1
       ) do
     # Filtering by subway and light_rail types
     with {:ok, routes_at_stop} <- fetch_routes_by_stop_fn.(stop_id, now, [0, 1]),
          route_ids_at_stop = Enum.map(routes_at_stop, & &1.route_id),
          {:ok, alerts} <- fetch_alerts_fn.(route_ids: route_ids_at_stop),
-         {:ok, stop_sequences} <- fetch_stop_sequences_by_stop_fn.(stop_id, route_ids_at_stop) do
+         {:ok, stop_sequences, platform_to_station_map} <-
```

Comment:
Can the data processing that happens on lines 32-40 be moved to `fetch_parent_station_sequences_through_stop` so that this function only returns the final iteration of `stop_sequences`?